### DISCUSSION TITLE
[MTR05] week4 關於 node.js 之錯誤訊息

### DISCUSSION THREAD

**Discussion by cmtilo** [Lidemy/forum#89] [程式導師實驗計畫]
請問目前正在練習發最基本的request時，有出現錯誤訊息，看不太懂，有人能分享經驗嗎??

以下疑問說明：
js檔輸入以下程式碼
```JS
const request = require('request');

request('http://www.google.com', function(error, response, body) {
  console.error('error:', error);
  console.log('statusCode:', response && response.statusCode);
  console.log('body:', body);
});

```
用node執行的時候出現一串錯誤訊息
```
C:\Users\cmt\mentor-program-5th-cmtilo>node hw1.js
internal/modules/cjs/loader.js:968
  throw err;
  ^

Error: Cannot find module 'C:\Users\cmt\mentor-program-5th-cmtilo\hw1.js'
[90m    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:965:15) [39m
[90m    at Function.Module._load (internal/modules/cjs/loader.js:841:27) [39m
[90m    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12) [39m
[90m    at internal/main/run_main_module.js:17:47 [39m {
  code:[32m'MODULE_NOT_FOUND'[39m,
  requireStack: []
}

```

上網找到[這篇](https://ithelp.ithome.com.tw/questions/10198535)也有提到類似的錯誤訊息，但他們的對話我看不太懂...

**Comment by aszx87410** [comment]
> Error: Cannot find module 'C:\Users\cmt\mentor-program-5th-cmtilo\hw1.js'

Can not find module，找不到
找不到： `C:\Users\cmt\mentor-program-5th-cmtilo\hw1.js`

你確定這檔案有存在嗎？

**Reply by cmtilo** [reply]
老師我更正路徑了，不過還是錯誤，補個迎面截圖畫面：[截圖畫面](https://static.coderbridge.com/img/cmtilo/962bac65b7b141c7b72f8c03d3b8970f.png)

**Reply by cmtilo** [reply]
老師我可以了，我查到[這篇](https://blog.csdn.net/xiejunna/article/details/71170542)，現在好像成功了：[截圖](https://static.coderbridge.com/img/cmtilo/6bff399dd7aa453b838c603970824c65.jpg)
不過不知道為什麼原本不行?現在可以..也不知道為什麼可以了?😅

**Reply by aszx87410** [reply]
錯誤訊息跟你說 request 這個模組找不到，所以你忘記 npm install request 了

**Comment by cmtilo** [comment]
提醒點進來這篇的各位...務必記得 `npm install request`
